### ISSUE TITLE
Build of ext-react that depends upon latest version of ScalaJS-React, please

### ISSUE THREAD

**Issue by djneades** [japgolly/scalacss#120]
Would it be possible to produce a build of `ext-react` that depends upon the current release of ScalaJS-React (v.1.1.0), please? The reason I am asking is to avoid library dependency warnings like the following in my SBT builds:

```
[warn] Found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[warn]
[warn]  * com.github.japgolly.scalajs-react:core_sjs0.6_2.12:1.1.0 is selected over 1.0.0
[warn]      +- com.github.japgolly.scalajs-react:ext-scalaz72_sjs0.6_2.12:1.1.0 (depends on 1.1.0)
[warn]      +- com.github.japgolly.scalajs-react:extra_sjs0.6_2.12:1.1.0 (depends on 1.1.0)
[warn]      +- com.araxis:mw_sjs0.6_2.12:2017.1                   (depends on 1.1.0)
[warn]      +- com.github.japgolly.scalajs-react:ext-cats_sjs0.6_2.12:1.1.0 (depends on 1.1.0)
[warn]      +- com.github.japgolly.scalajs-react:ext-monocle_sjs0.6_2.12:1.1.0 (depends on 1.1.0)
[warn]      +- com.github.japgolly.scalacss:ext-react_sjs0.6_2.12:0.5.3 (depends on 1.0.0)
[warn]
[warn] Run 'evicted' to see detailed eviction warnings
```

Many thanks in advance!

**Comment by japgolly** [comment]
I'm doing this in the next release but I don't know of a long-term solution. They are actually compatible and nearly always will be because the surface area of sjr that this touches is tiny; the eviction warning isn't indicating a real issue, it just looks at the version numbers. It's unreasonable to make a new ScalaCSS everytime I release scalajs-react so I'm not sure what to do here.

**Comment by dneades** [comment]
@japgolly Thank you, whatever you can do is appreciated. Going forward, I realize that he eviction warning is (usually) going to be benign, but it is nevertheless a significant amount of noise when one is aiming for a clean (i.e. warning-free) build. Perhaps one answer is to build ScalaCSS from source in projects that use it, though that too would not be ideal.

**Comment by japgolly** [comment]
I think there are two possible ways out of this:

1. Get the annoyingly inflexible behaviour changed in SBT. It seems that it's already under consideration.
2. If someone were to contribute a PR that would allow performing releases from Travis then I could hand out release authority and someone else could do a release...? Or something similar. I don't know, just a thought.

@dneades How do you aim for a warning-free build with the newer versions of SBT (1.x)? They're way worse. They spam me about all kinds of evictions and I don't seem to be able to do anything about it. It's incredibly frustrating.

**Comment by dneades** [comment]
@japgolly I am no expert in the Scala ecosystem, so take the following thoughts for what they are worth (i.e. probably not much).

Trying to minimize eviction warnings can sometimes feel like a rather Sisyphean task, especially since one is utterly at the mercy of the maintainers of the project’s library dependencies. All I am really able to do is to try to minimize external dependencies, and attempt to keep everything reasonably current. I am having to pick-and-choose when I upgrade particular dependencies, waiting for those opportune times when the common dependencies of two or more libraries align.

Part of the problem is perhaps that much of the Scala ecosystem is still at a fairly early, fast-moving stage of development, where binary APIs are not necessarily locked-down. Some things are beginning to mature, so the situation may improve. Cats hitting 1.x is a big deal, for example, and the work towards Monix 3.x is beginning to settle down. Still, it’s hard not to view the diamond-dependency problem as a significant headache for large Scala projects. (Indeed, even without _that_ problem, it can feel like a significant amount of work just keeping one’s code compiling and working with the fast-changing APIs of external libraries.)

Right now, my project is at a fairly early stage and its dependency list isn’t _that_ large. Thus, the only evictions I am seeing for my build are due to the `org.webjars.npm` React 16.2.0 dependencies (`react` and `react-dom`) that I have. Using those results in this:

```
Found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[warn]  * org.webjars.npm:object-assign:4.1.1 is selected over [4.1.0,5)
[warn]      +- org.webjars.npm:react:16.2.0                       (depends on 4.1.1)
[warn]      +- org.webjars.npm:react-dom:16.2.0                   (depends on 4.1.1)
[warn]      +- org.webjars.npm:prop-types:15.6.0                  (depends on 4.1.1)
[warn]      +- org.webjars.npm:fbjs:0.8.16                        (depends on [4.1.0,5))
[warn]  * org.webjars.npm:loose-envify:1.3.1 is selected over {[1.3.1,2), [1.0.0,2)}
[warn]      +- org.webjars.npm:react:16.2.0                       (depends on 1.3.1)
[warn]      +- org.webjars.npm:react-dom:16.2.0                   (depends on 1.3.1)
[warn]      +- org.webjars.npm:fbjs:0.8.16                        (depends on [1.0.0,2))
[warn]      +- org.webjars.npm:prop-types:15.6.0                  (depends on [1.3.1,2))
[warn] Run 'evicted' to see detailed eviction warnings
```

At first glance, that looks like a genuine issue with those webjars, and I was rather hoping that, when ScalaJS-React eventually moves to React 16.x, it would have a better source for that than I was able to find – one without the conflicting dependencies :-) (I did try depending directly on the React NPM modules via `scalajs-bundler`, but that brought with it significant pain, so I reverted back to the webjars.)
